addons/stock/models/stock_quant.py

```diff
@@ -293,7 +293,7 @@ def action_view_inventory(self):
 
     def action_apply_inventory(self):
         # Update the context to prevent recursive call from write method
-        self = self.with_context({'inventory_report_mode': False})
+        self = self.with_context(inventory_report_mode=False)
```

Comment:
Hello! I started looking into this but didn't finish before you made this PR :smiling_face_with_tear:

I believe we can delete this line (and the comment above it) because it was added here: https://github.com/odoo/odoo/pull/83884 with the explanation that it's due to the write being called too many times here: https://github.com/odoo/odoo/blob/8b155b695823dfa954464ad0d5dd07445ece2471/addons/stock/models/stock_quant.py#L244-L248 but then this later PR: https://github.com/odoo/odoo/pull/84771 deletes the same problem lines.

TLDR: You might be able to delete this line (it doesn't hurt to leave it in, it just might end up being dead code) + a test would be nice too so the feature doesn't break again :smiley:

Reply:
Hello Tiffany
Yes, Indeed, sorry I didn't notice that. I added a test and your explanation in the commit message.
Thank you for your help ;)